integration/store_gateway_limits_hit_test.go

```diff
@@ -0,0 +1,160 @@
+				"-blocks-storage.tsdb.block-ranges-period":          blockRangePeriod.String(),
+				"-blocks-storage.tsdb.head-compaction-idle-timeout": "1s",
+				"-blocks-storage.tsdb.retention-period":             ((blockRangePeriod * 2) - 1).String(),
+				"-blocks-storage.tsdb.ship-interval":                "1s",
+			},
+		)
+
+		consul := e2edb.NewConsul()
+		minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+		require.NoError(t, scenario.StartAndWaitReady(consul, minio))
+
+		// Start Mimir components.
+		distributor := e2emimir.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), flags)
+
+		ingester = e2emimir.NewIngester("ingester", consul.NetworkHTTPEndpoint(), flags)
+
+		additionalStoreGatewayFlags = mergeFlags(flags, additionalStoreGatewayFlags)
+		storeGateway = e2emimir.NewStoreGateway("store-gateway", consul.NetworkHTTPEndpoint(), additionalStoreGatewayFlags)
+
+		require.NoError(t, scenario.StartAndWaitReady(distributor, ingester, storeGateway))
+
+		additionalQuerierFlags = mergeFlags(flags, additionalQuerierFlags)
+		querier := e2emimir.NewQuerier(querierTag, consul.NetworkHTTPEndpoint(), additionalQuerierFlags)
+
+		require.NoError(t, scenario.StartAndWaitReady(querier))
+
+		client, err = e2emimir.NewClient(distributor.HTTPEndpoint(), querier.HTTPEndpoint(), "", "", orgID)
+		require.NoError(t, err)
+		return
+	}
+
+	tests := map[string]struct {
+		additionalStoreGatewayFlags map[string]string
+		additionalQuerierFlags      map[string]string
+		expectedErrorKey            string
+	}{
+		"when store-gateway hits max_fetched_series_per_query, 'exceeded series limit' is returned": {
+			additionalStoreGatewayFlags: map[string]string{"-querier.max-fetched-series-per-query": "3"},
+			expectedErrorKey:            storegateway.ErrSeriesLimitMessage,
+		},
+		"when querier hits max_fetched_series_per_query, 'err-mimir-max-series-per-query' is returned": {
+			additionalQuerierFlags: map[string]string{"-querier.max-fetched-series-per-query": "3"},
+			expectedErrorKey:       string(globalerror.MaxSeriesPerQuery),
+		},
+		"when store-gateway hits max_fetched_chunks_per_query, 'exceeded chunks limit' is returned": {
+			additionalStoreGatewayFlags: map[string]string{"-querier.max-fetched-chunks-per-query": "3"},
+			expectedErrorKey:            storegateway.ErrChunksLimitMessage,
+		},
+		"when querier hits max_fetched_chunks_per_query, 'err-mimir-max-chunks-per-query' is returned": {
+			additionalQuerierFlags: map[string]string{"-querier.max-fetched-chunks-per-query": "4"},
+			expectedErrorKey:       string(globalerror.MaxChunksPerQuery),
+		},
+	}
+
+	for testName, testData := range tests {
+		t.Run(testName, func(t *testing.T) {
+			scenario, ingester, storeGateway, client := setup(t, testData.additionalStoreGatewayFlags, testData.additionalQuerierFlags)
+			defer scenario.Close()
+
+			timeStamps12, timeSeries12 := createTimeSeries(t, []string{series1, series2})
```

Comment:
[nit] Can we create all timeseries upfront here, instead of calling `createTimeSeries()` twice (here and below)? If yes, then what's the need of `createTimeSeries()` at all?